### PR TITLE
Added es6 export default function module support.

### PR DESCRIPTION
**Pull request by dbillingham**
Within my required files before running cucumber I have to export using the commonjs syntax:
```
module.exports = function() {
  this.World = World;
};
```

I would like to export these using the es6 module syntax:
```
export default function() {
  this.World = World;
}
```

Babel 6 now correctly transpiles this to not include:
```
module.exports = exports["default"];
``` 

I would like to be able to require es6 modules without the need of a babel plugin to incorrectly add the module.exports to the bottom of the transpiled file.

This pr contains some code for discussion to achieve this.

### CODE REVIEW

File: lib/cucumber/cli/support_code_loader.js [hunk 1]
```diff
@@ -61,8 +61,13 @@ function SupportCodeLoader(supportCodeFilePaths, compilerModules) {
         var supportCodeHelper = this;
         paths.forEach(function (path) {
           var initializer = require(path);
-          if (typeof(initializer) === 'function')
+          if (typeof(initializer) === 'function') {
             initializer.call(supportCodeHelper);
+          } else if (initializer.hasOwnProperty('default')) {
+            if (typeof(initializer.default) === 'function') {
```
Comment (charlierudolph) [inline lib/cucumber/cli/support_code_loader.js]: can we join the two conditions?
```js
else if (initializer.hasOwnProperty('default') && typeof(initializer.default) === 'function') {
  initializer.default.call(supportCodeHelper);
}
```

**Comment by charlierudolph** [comment]
Sorry for the delay. Just the one comment and then looks good to me

**Comment by charlierudolph** [comment]
@dbillingham bump

**Comment by dbillingham** [comment]
Thanks @charlierudolph, pr has been updated.

**Comment by charlierudolph** [comment]
Thanks for the contribution! I should be releasing a new version this weekend that will include this

**Pull request merged**